onnxruntime/core/session/inference_session.cc

```diff
@@ -97,41 +98,175 @@ inline std::basic_string<T> GetCurrentTimeString() {
 
 std::atomic<uint32_t> InferenceSession::global_session_id_{1};
 
-InferenceSession::InferenceSession(const SessionOptions& session_options,
-                                   logging::LoggingManager* logging_manager)
-    : session_options_(session_options),
-      graph_transformation_mgr_(session_options.max_num_graph_transformation_steps),
-      logging_manager_(logging_manager),
+static Status FinalizeSessionOptions(const SessionOptions& user_provided_session_options,
+                                     const ONNX_NAMESPACE::ModelProto* model_proto,
+                                     /*out*/ SessionOptions& finalized_session_options) {
+  const logging::Logger& default_logger = logging::LoggingManager::DefaultLogger();
```

Comment:
Could you avoid to use the default logger?

Reply:
We can't avoid using the default logger. It is the only option available since the session_logger_ cannot be initialized until the session options have been finalized...